circuits/offchain/credentialAtomicQueryV3OffChain.circom

```diff
@@ -0,0 +1,269 @@
+    signal input requestID;
+    signal input userGenesisID;
+    signal input profileNonce;
+    signal input claimSubjectProfileNonce; // nonce of the profile that claim is issued to, 0 if claim is issued to genesisID
+
+    signal input issuerID;
+    signal input isRevocationChecked;
+    signal input issuerClaimNonRevMtp[issuerLevels];
+    signal input issuerClaimNonRevMtpNoAux;
+    signal input issuerClaimNonRevMtpAuxHi;
+    signal input issuerClaimNonRevMtpAuxHv;
+    signal input issuerClaimNonRevClaimsTreeRoot;
+    signal input issuerClaimNonRevRevTreeRoot;
+    signal input issuerClaimNonRevRootsTreeRoot;
+    signal input issuerClaimNonRevState;
+    /* current time */
+    signal input timestamp;
+
+    /** Query */
+    signal input claimSchema;
+
+    signal input claimPathNotExists; // 0 for inclusion, 1 for non-inclusion
+    signal input claimPathMtp[claimLevels];
+    signal input claimPathMtpNoAux; // 1 if aux node is empty, 0 if non-empty or for inclusion proofs
+    signal input claimPathMtpAuxHi; // 0 for inclusion proof
+    signal input claimPathMtpAuxHv; // 0 for inclusion proof
+    signal input claimPathKey; // hash of path in merklized json-ld document
+    signal input claimPathValue; // value in this path in merklized json-ld document
+
+    signal input slotIndex;
+    signal input operator;
+    signal input value[valueArraySize];
+
+    signal input issuerClaim[8];
+
+    // Diff for MTP proof
+    signal input issuerClaimMtp[issuerLevels];
+    signal input issuerClaimClaimsTreeRoot;
+    signal input issuerClaimRevTreeRoot;
+    signal input issuerClaimRootsTreeRoot;
+    signal input issuerClaimIdenState;
+
+    // Diff for Sig proof
+    signal input issuerAuthClaim[8];
+    signal input issuerAuthClaimMtp[issuerLevels];
+    signal input issuerAuthClaimsTreeRoot;
+    signal input issuerAuthRevTreeRoot;
+    signal input issuerAuthRootsTreeRoot;
+    signal input issuerAuthClaimNonRevMtp[issuerLevels];
+    signal input issuerAuthClaimNonRevMtpNoAux;
+    signal input issuerAuthClaimNonRevMtpAuxHi;
+    signal input issuerAuthClaimNonRevMtpAuxHv;
+    signal input issuerClaimSignatureR8x;
+    signal input issuerClaimSignatureR8y;
+    signal input issuerClaimSignatureS;
+
+    // Diif for Sig proof output
+    signal output issuerAuthState;
+    
+     /*
+    >>>>>>>>>>>>>>>>>>>>>>>>>>> End Inputs <<<<<<<<<<<<<<<<<<<<<<<<<<<<
+    */
+
+    /////////////////////////////////////////////////////////////////
+
+    // Check issuerClaim is issued to provided identity
+    verifyCredentialSubjectProfile()(
+        issuerClaim,
+        userGenesisID,
+        claimSubjectProfileNonce
+    );
+
+    // Verify issuerClaim schema
+    verifyCredentialSchema()(1, issuerClaim, claimSchema);
+
+    // verify issuerClaim expiration time
+    verifyExpirationTime()(issuerClaim, timestamp);
+    
+    signal isSig;
+    signal isMTP;
+    isSig <-- proofType==0 ? 1 : 0;
+    isMTP <-- proofType==1 ? 1 : 0;
```

Comment:
May be this works fine.
However <-- is not really a constraint.
The following should work for sure and prevent from enabling both MTP and SIG at a time
```
isSig  <== 1 - proofType;
isMTP <== proofType;
isSig * isMTP === 0;
```

Reply:
done